save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkContestProjectController.kt

```diff
@@ -144,6 +144,12 @@ class LnkContestProjectController(
         Mono.justOrEmpty<Project>(projectService.findByNameAndOrganizationName(projectName, organizationName)),
         Mono.justOrEmpty(contestService.findByName(contestName)),
     )
+        .switchIfEmpty {
+            Mono.error(ResponseStatusException(HttpStatus.NOT_FOUND))
+        }
+        .filter { (project, _) ->
+            project.public
+        }
         .switchIfEmpty {
             Mono.error(ResponseStatusException(HttpStatus.NOT_FOUND))
```

Comment:
If you use `findWithPermission` in the start of the method, that by this point we can assume that the user has at least read access to the project and we can send a more meaningful response, e.g. `409 Conflict: Private projects cannot be enrolled in the contest`

Reply:
Yeah, but why at first check `READ` and then filter unless `WRITE`?

Reply:
In order to provide better error code and message